File: socialNetwork/src/main/java/com/ftn/socialNetwork/model/entity/Image.java
package com.ftn.socialNetwork.model.entity;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "images")
public class Image {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column
    private String path;
    @ManyToOne
    @JsonIgnore
    private Post post;
    @OneToOne
    @JsonIgnore
    private User user;


}

File: src/app/models/image.ts
import { User } from "./user";
import { Post } from "./post";

export class Image {
    id: number;
    path: string;
    user: User;
    post: Post;
    
    constructor(id: number, path: string, user: User, post: Post) {
      this.id = id;
      this.path = path;
      this.user = user;
      this.post = post;
    }
  }
  